extensions/amp-story/1.0/story-analytics.js

```diff
@@ -76,26 +76,40 @@ export class StoryAnalyticsService {
 
     /** @const @private {!./variable-service.AmpStoryVariableService} */
     this.variableService_ = getVariableService(win);
+
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = getStoreService(win);
+
+    this.initializeListeners_();
   }
 
-  /**
-   * @param {!./navigation-state.StateChangeEventDef} stateChangeEvent
-   */
-  onNavigationStateChange(stateChangeEvent) {
-    switch (stateChangeEvent.type) {
-      case StateChangeType.ACTIVE_PAGE:
+  /** @private */
+  initializeListeners_() {
+    this.storeService_.subscribe(StateProperty.BOOKEND_STATE, isActive => {
+      this.triggerEvent(
+        isActive ? AnalyticsEvent.BOOKEND_ENTER : AnalyticsEvent.BOOKEND_EXIT
+      );
+    });
+
+    this.storeService_.subscribe(
+      StateProperty.CURRENT_PAGE_ID,
+      pageId => {
+        if (pageId.length <= 0) {
```

Comment:
Is there any reason why you don't want to test `!pageId`?
This would throw an error for `null` or `undefined`, which isn't used today, but this test makes it look like `pageId` is an array

Reply:
Since `CURRENT_PAGE_ID` is initialized as '' in the store, I thought it might be safer. But both work fine. Changed it to `!pageId`.